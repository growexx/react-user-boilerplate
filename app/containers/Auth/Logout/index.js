import React from 'react';
import StorageService from 'utils/StorageService';
import { EMITTER_EVENTS } from '../../../utils/constants';
import Emitter from '../../../utils/events';

const Logout = () => {
  StorageService.clear();
  Emitter.emit(EMITTER_EVENTS.LOG_OUT);
  return <React.Fragment />;
};

export default Logout;
