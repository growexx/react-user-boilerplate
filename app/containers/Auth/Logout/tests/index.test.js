/**
 *
 * Tests for Registration
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import history from 'utils/history';
import { browserHistory } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import configureStore from '../../../../configureStore';
import Logout from '../index';
import Lodable from '../Loadable';
let store;
const componentWrapper = Component =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <Component />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

describe('<Registration />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper(Logout);
    expect(firstChild).toMatchSnapshot();
  });
  it('Should render and match the snapshot Loadable', () => {
    const {
      container: { firstChild },
    } = componentWrapper(Lodable);
    expect(firstChild).toMatchSnapshot();
  });
});
