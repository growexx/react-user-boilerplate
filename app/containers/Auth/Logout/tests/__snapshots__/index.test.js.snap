// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Registration /> Should render and match the snapshot 1`] = `null`;

exports[`<Registration /> Should render and match the snapshot Loadable 1`] = `null`;
